packages/flutter/lib/src/material/input.dart

```diff
@@ -137,6 +147,15 @@ class _InputState extends State<Input> {
       config.onSubmitted(_value);
   }
 
+  void _handleSelectionChanged(TextSelection selection) {
+    if (_isAttachedToKeyboard) {
+      _keyboardHandle.setSelection(selection.start, selection.end);
```

Comment:
Ok, I think I know what the problem really is. It's that we have the keyboard logic in the Material layer instead of in `widgets/editable.dart`. But we can fix that in a later patch.